StarRailBattleSim/src/characters/yunli/YunliEmergancyUlt.java

```diff
@@ -0,0 +1,24 @@
+package characters.yunli;
+
+import battleLogic.AbstractEntity;
+import characters.goal.UltGoal;
+
+public class YunliEmergancyUlt extends UltGoal<Yunli> {
+    public YunliEmergancyUlt(Yunli character) {
+        super(character);
+    }
+
+    @Override
+    public UltGoalResult determineAction() {
```

Comment:
As I understood her ult, she needs someone to take a turn, so if she ults at the very end. It won't do anything, this way we ult before the last turn is taken (regardless of enemy or not) and get an extra ult in.

I don't have her, so can't really check if this is how it works; but that was my thought process here

Reply:
Ah yes, that is how Yunli works. However the getBattle().isAboutToEnd() method checks if the next character to move is outside the battle's remaining AV, which means checking the 2nd next person to move won't do anything since currently that logic would only trigger when no more turns are remaining. 

I think what this goal should do instead is check that `getUnit(0)` is inside the battle's remaining AV, but that `getUnit(1)` is outside the battle's remaining AV, and then return DO in that case. And then you can just get rid of the `isAboutToEnd` check since it's not very useful for Yunli's case.

Reply:
Made the changes, didn't change any dmg values at 550 AV. Does on 500 AV